### DISCUSSION TITLE
Illuminate\Support\Collection<int,mixed>::get() expects int, string given

### DISCUSSION THREAD

**Discussion by johanmolen** [larastan/larastan#1200] [Q&A] [answered]
Does somebody know if the following error is related to my code, Laravel or Larastan?
![image](https://user-images.githubusercontent.com/8382746/160089919-f20d09c3-a6ae-4e97-b26e-890a1c87fbe2.png)

I have a collection which I use `pluck()` on, that way I get a new collection with a string as key and string as value.
Then I use the `get()` method on the collection to retrieve a value based on the string.

laravel/framework is on version 9.5.1
nunomaduro/larastan is on version 2.1.3

**Comment by szepeviktor** [comment, marked as answer]
Hello Johan! 👋 
People around here like to see your source code, at least 1 line.
`Collection` is generic.
https://github.com/nunomaduro/larastan/blob/17d72fdeb3d4f6ecfb3cc043daac8df766c2c77a/stubs/Collection.stub#L9-L16
Try specifying key and value type in your code.

**Reply by johanmolen** [reply]
@szepeviktor thanks! I fixed it by adding this:
```php
/** @var Collection<string,string> $additionalFields */
$additionalFields = collect(data_get($shippingAddress, 'CUSTOM_FIELDS', []))
            ->pluck('VALUE', 'KEY');
```

**Reply by johanmolen** [reply]
@szepeviktor you are the best, forgot to mention it!
